Feature stock assignment inventory by code

lomamech opened the pull request:
feature(Stock Assignment Inventory select by Code)
- Use component for select inventory
- Update E2E test by the using of component inventorySelect

lomamech committed:
feature(Stock Assignment Inventory select by Code)
- Use component for select inventory

closes #4256

lomamech committed:
test(Assignment Modal inventory select display code)
- Update E2E test by the using of component inventorySelect

closes #4256

jniles commented on test/end-to-end/stock/stock.assign.spec.js:
```diff
@@ -128,4 +128,4 @@ function StockAssignTests() {
 }
 
-describe('Stock Assign Module', StockAssignTests);
+describe.only('Stock Assign Module', StockAssignTests);
```
This needs to be removed - it is only testing these tests.

lomamech replied:
Sorry

lomamech committed:
text(Stock Assignment Inventory Code)
- Sanitaze code, remove attribut only in E2E test for Stock Assignment
  inventory

jniles reviewed (dismissed):
@lomamech I just realized this needs to use `$ctrl.availableInventories`.  It shouldn't use all inventory, but only those that are available.  `bhInventorySelect` might not be able to do that.

One thing you could do is make `bhInventorySelect` take an optional parameter `depot-uuid` that will then filter based on inventory in the depot.

lomamech committed:
improvement(Stock Assignment)
- Restoration of uiSelect for the selection of inventories
- Addition of code display for selection
- Form reset when changing depots or inventories
- Display of the available quantity and prevention of assignment of a
  quantity greater than the available stock

jniles commented on client/src/modules/stock/assign/modals/action.modal.html:
```diff
@@ -25,17 +25,17 @@
         </ui-select-match>
-        <ui-select-choices ui-select-focus-patch repeat="i.uuid as i in $ctrl.availableInventories | filter: { 'text': $select.search }">
-          <span ng-bind-html="i.text | highlight:$select.search"></span>
+        <ui-select-choices ui-select-focus-patch repeat="i.uuid as i in $ctrl.availableInventories | filter: { 'hrLabel': $select.search }">
```
Excellent 👍

jniles commented on client/src/modules/stock/assign/modals/action.modal.html:
```diff
@@ -76,10 +76,13 @@
 
     <!-- quantity -->
-    <div class="form-group">
+    <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.quantity.$invalid }">
```
Nice 👍  Good catch!

jniles commented on client/src/modules/templates/bhInventorySelect.tmpl.html:
```diff
@@ -12,6 +12,7 @@
       name="inventory_uuid"
       ng-model="$ctrl.inventoryUuid"
       on-select="$ctrl.onSelect($item, $model)"
+      ng-disabled="$ctrl.ngDisabled"
```
Shouldn't this be removed?  Where did this come from?

jniles reviewed (commented):
@lomamech, this looks good to me.  I've made one comment that I think is just an accident.  You left something in `bhInventorySelect` even though you didn't use it.

lomamech committed:
Remove not necessary params

jniles reviewed (approved):
Perfect :+1:

bors r+
